File: python/src/acp_sdk/models/schemas.py
from pydantic import BaseModel

from acp_sdk.models.models import Agent, AgentName, AwaitResume, Event, Message, Run, RunMode, SessionId


class PingResponse(BaseModel):
    pass


class AgentsListResponse(BaseModel):
    agents: list[Agent]


class AgentReadResponse(Agent):
    pass


class RunCreateRequest(BaseModel):
    agent_name: AgentName
    session_id: SessionId | None = None
    input: list[Message]
    mode: RunMode = RunMode.SYNC


class RunCreateResponse(Run):
    pass


class RunResumeRequest(BaseModel):
    await_resume: AwaitResume
    mode: RunMode


class RunResumeResponse(Run):
    pass


class RunReadResponse(Run):
    pass


class RunCancelResponse(Run):
    pass


class RunEventsListResponse(BaseModel):
    events: list[Event]

File: typescript/src/models/schemas.ts
import * as z from "zod";
import {
  Agent,
  AgentName,
  AwaitResume,
  Event,
  Message,
  Run,
  RunMode,
  SessionId,
} from "./models.js";

export const PingResponse = z.object({});

export const AgentsListResponse = z.object({
  agents: z.array(Agent),
});

export const AgentsReadResponse = Agent;

export const RunCreateRequest = z.object({
  agent_name: AgentName,
  session_id: z.optional(SessionId),
  input: z.array(Message),
  mode: RunMode,
});

export type RunCreateRequest = z.infer<typeof RunCreateRequest>;

export const RunCreateResponse = Run;

export const RunEventsListResponse = z.object({
  events: z.array(Event),
});

export const RunResumeRequest = z.object({
  await_resume: AwaitResume,
  mode: RunMode,
})

export type RunResumeRequest = z.infer<typeof RunResumeRequest>;

export const RunResumeResponse = Run;

export const RunReadResponse = Run;

